### PR TITLE
Get the actual error message from Dremio

### PR DESCRIPTION
**Pull request by ravjotbrar**
### Summary

Prior to this change, we were not checking to see of a job is in a FAILED state. Instead, we would attempt to get job_results for every job, which would return an error about not being able to get results of a job in a failed state. Now we check to see if the job failed and return the error message in the payload.

### Description

In cursor.py:

```
            if job_status_state == "FAILED":
                error_message = job_status_response["errorMessage"]
                raise Exception(f"ERROR: {error_message}")
```

#### Unit Tests
tests/unit/test_connection.py .                                                               [ 50%]
tests/unit/test_payload_error.py .                                                            [100%]

================================= 2 passed in 1.08s ===================================

#### Grants Functional Tests
tests/functional/adapter/grants/test_incremental_grants.py .             [ 16%]
tests/functional/adapter/grants/test_invalid_grants.py .                 [ 33%]
tests/functional/adapter/grants/test_model_grants.py ..                  [ 66%]
tests/functional/adapter/grants/test_seed_grants.py .                    [ 83%]
tests/functional/adapter/grants/test_snapshot_grants.py .                [100%]

================== 6 passed, 18 warnings in 586.39s (0:09:46) ==================

#### Basic Functional Tests
tests/functional/adapter/basic/test_adapter_methods.py .                 [  7%]
tests/functional/adapter/basic/test_base_mat.py .                        [ 14%]
tests/functional/adapter/basic/test_docs_generate.py ...                 [ 35%]
tests/functional/adapter/basic/test_empty.py .                           [ 42%]
tests/functional/adapter/basic/test_ephemeral.py .                       [ 50%]
tests/functional/adapter/basic/test_generic_tests.py .                   [ 57%]
tests/functional/adapter/basic/test_incremental.py ..                    [ 71%]
tests/functional/adapter/basic/test_singular_ephemeral.py .              [ 78%]
tests/functional/adapter/basic/test_singular_tests.py .                  [ 85%]
tests/functional/adapter/basic/test_snapshots.py s.                      [100%]

================== 13 passed, 1 skipped, 35 warnings in 607.06s (0:10:07) ==================

### Changelog

-   [x] Added a summary of what this PR accomplishes to CHANGELOG.md

### Related Issue

https://github.com/dremio/dbt-dremio/issues/69

### CODE REVIEW

File: tests/functional/adapter/grants/test_invalid_grants.py [hunk 1]
```diff
@@ -18,11 +18,10 @@
 from tests.utils.util import relation_from_name
 from dbt.tests.util import get_connection
 
-# Currently we return an HTTP error, but need to improve error handling
-# to get a better error message (https://github.com/dremio/dbt-dremio/issues/69)
+
 class TestInvalidGrantsDremio(BaseGrantsDremio, BaseInvalidGrants):
     def grantee_does_not_exist_error(self):
-        return "Can not fetch details for a job that is in [FAILED] state"
+        return "StatusRuntimeException: INTERNAL"
```
Comment (ArgusLi) [inline tests/functional/adapter/grants/test_invalid_grants.py:24]: These error messages do not correspond to their functions. Is it possible to align the error messages to the function names?

**Review by ArgusLi — COMMENTED** [review]
Apart from this small nitpick, looks good to me :)

File: tests/functional/adapter/grants/test_invalid_grants.py (hunk 1 above)
Comment (ArgusLi) [reply tests/functional/adapter/grants/test_invalid_grants.py:24]: Sounds good, thank you!

**Review by jlarue26 — APPROVED** [review]
Looks great, thx!

**Pull request merged**